subsys/net/lib/http/http_server_http1.c

```diff
@@ -71,6 +71,39 @@ static int send_http1_409(struct http_client_ctx *client)
 				       sizeof(conflict_response) - 1);
 }
 
+static void send_http1_500(struct http_client_ctx *client, int error_code)
+{
+#define HTTP_500_RESPONSE_TEMPLATE			\
+	"HTTP/1.1 500 Internal Server Error\r\n"	\
+	"Content-Type: text/plain\r\n"			\
+	"Content-Length: %d\r\n\r\n"			\
+	"Internal Server Error: %s\r\n"
+#define MAX_ERROR_DESC_LEN 32
+
+	/* Placeholder for the error description. */
+	char error_str[] = "xxx";
+	char http_response[sizeof(HTTP_500_RESPONSE_TEMPLATE) +
+			   sizeof("xx") + /* Content-Length */
+			   MAX_ERROR_DESC_LEN + 1]; /* For the error description */
+	const char *error_desc;
+	int desc_len;
+
+	/* Try to fetch error description, fallback to error number if not available */
+	error_desc = strerror(error_code);
+	if (strlen(error_str) == 0) {
+		/* Cast error value to uint8_t to avoid truncation warnings. */
+		(void)snprintk(error_str, sizeof(error_str), "%u", (uint8_t)error_code);
+		error_desc = error_str;
+	}
+	desc_len = MIN(MAX_ERROR_DESC_LEN, strlen(error_desc));
+
+	(void)snprintk(http_response, sizeof(http_response),
+		       HTTP_500_RESPONSE_TEMPLATE,
+		       strlen("Internal Server Error: \r\n") + desc_len,
```

Comment:
perhaps `sizeof` here to avoid runtime call.

Reply:
Updated, thanks.